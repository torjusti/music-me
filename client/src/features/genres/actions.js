export const setGenreSelected = (genre, selected) => ({
  type: 'SET_GENRE_SELECTED',
  payload: {
    genre,
    selected,
  },
});
