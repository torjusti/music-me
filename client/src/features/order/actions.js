/**
 * Sets the column on which we are sorting.
 */
export const setColumn = orderBy => ({
  type: 'SET_COLUMN',
  payload: { orderBy },
});

/**
 * Toggles the sorting direction on the currently selected column.
 */
export const toggleDirection = () => ({
  type: 'TOGGLE_DIRECTION',
});

/**
 * Signifies that we should stop sorting on the currently selected column.
 */
export const clearOrder = () => ({
  type: 'CLEAR_ORDER',
});
