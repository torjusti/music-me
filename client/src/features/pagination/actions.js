/**
 * Sets the pagination page which is being shown in the table.
 */
export const setPage = page => ({
  type: 'SET_PAGE',
  payload: { page },
});
