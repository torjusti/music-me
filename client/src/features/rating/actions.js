export const setRatingSelected = selectedRating => ({
  type: 'SET_SELECTED_RATING',
  payload: {
    selectedRating,
  },
});

export const toggleRatingEnabled = () => ({
  type: 'TOGGLE_RATING_ENABLED',
});
