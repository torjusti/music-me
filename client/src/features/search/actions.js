export const setQuery = query => ({
  type: 'SET_QUERY',
  payload: { query },
});

export const clearQuery = () => ({
  type: 'CLEAR_QUERY',
});
