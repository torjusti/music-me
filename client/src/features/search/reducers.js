const search = (state = null, action) => {
  switch (action.type) {
    case 'SET_QUERY':
      return action.payload.query;

    case 'CLEAR_QUERY':
      return null;

    default:
      return state;
  }
};

export default search;
