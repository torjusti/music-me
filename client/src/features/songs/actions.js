/**
 * Action which eventually sends a rating for a song to the server.
 */
export const sendSongRating = (id, rating) => ({
  type: 'SEND_SONG_RATING',
  payload: { id, rating },
});
