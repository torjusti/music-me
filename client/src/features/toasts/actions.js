export const showToast = message => ({
  type: 'SHOW_TOAST',
  payload: { message },
});
