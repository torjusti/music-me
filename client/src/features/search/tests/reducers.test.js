import search from '../reducers';
import * as actions from '../actions';

describe('search reducer', () => {
  it('should be null initially', () => {
    expect(search(undefined, {})).toEqual(null);
  });

  it('should set query', () => {
    expect(
      search(
        undefined,
        actions.setQuery(
          'test',
        ),
      ),
    ).toMatchSnapshot();
  });

  it('should clear query', () => {
    expect(
      search(
        undefined,
        actions.clearQuery(

        ),
      ),
    ).toMatchSnapshot();
  });

});
