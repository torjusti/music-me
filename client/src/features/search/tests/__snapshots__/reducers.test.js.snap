// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer should clear query 1`] = `null`;

exports[`search reducer should set query 1`] = `"test"`;
